#!/usr/bin/env python
PACKAGE = "moma_cartesian_impedance_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 1000, 0, 2000)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 100, 0, 200)
gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0.5, 0, 100)

gen.add("translational_damping_ratio", double_t, 0, "Cartesian translational damping", 1, 0, 10)
gen.add("rotational_damping_ratio", double_t, 0, "Cartesian rotational damping", 1, 0, 10)

gen.add("max_force", double_t, 0, "max force applied by the impedance controller [N]", 15, 0, 50)
gen.add("max_torque", double_t, 0, "max torque applied by the impedance controller [Nm]", 5, 0, 10)

gen.add("translational_stiffness_i", double_t, 0, "Cartesian translational stiffness i gain [N/(m*s)]", 5.0, 0, 100.0)
gen.add("rotational_stiffness_i", double_t, 0, "Cartesian rotational stiffness i gain [Nm/(rad*s)]", 0.5, 0, 10.0)

gen.add("translational_windup_limit", double_t, 0, "windup limit translation [N]", 10, 0, 50)
gen.add("rotational_windup_limit", double_t, 0, "windup limit rotation [Nm]", 5, 0, 10)
gen.add("reset_integrator_threshold", double_t, 0, "reset integerator if sign between error and integrator is negative and difference is larger than this threshold [N]", 2.0, 0, 10)



exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param"))
